File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int r; 
        long sum=0;
        while(x!=0){
            r=x%10;
            sum=sum*10+r;
            x=x/10;
        }
        if (sum < -2147483648 || sum > 2147483647) {
            return 0;
        }
        return (int)sum;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign=-1 if x<0 else 1
        x=abs(x)
        summ=0
        while(x!=0):
            summ=summ*10+x%10
            x//=10
        if summ<-2**31 or summ>2**31-1: return 0
        return sign*summ